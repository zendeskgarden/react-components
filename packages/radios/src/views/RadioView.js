/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import styled from 'styled-components';
import classNames from 'classnames';
import { retrieveTheme, isRtl } from '@zendeskgarden/react-theming';
import CheckboxStyles from '@zendeskgarden/css-forms/dist/checkbox.css';

const COMPONENT_ID = 'radios.radio_view';

/**
 * Used as a layout wrapper for other Radio views. Accepts all `<div>` props
 */
const RadioView = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  className: props =>
    classNames(CheckboxStyles['c-chk'], {
      // RTL
      [CheckboxStyles['is-rtl']]: isRtl(props)
    })
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default RadioView;
