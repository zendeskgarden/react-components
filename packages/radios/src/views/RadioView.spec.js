/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { render, renderRtl } from 'garden-test-utils';

import RadioView from './RadioView';

describe('RadioView', () => {
  it('renders default styling', () => {
    const { container } = render(<RadioView />);

    expect(container.firstChild).toHaveClass('c-chk');
  });

  it('renders RTL styling', () => {
    const { container } = renderRtl(<RadioView />);

    expect(container.firstChild).toHaveClass('is-rtl');
  });
});
