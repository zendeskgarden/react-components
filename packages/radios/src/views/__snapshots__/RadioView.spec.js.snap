// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioView renders RTL styling 1`] = `
<div
  className="c-chk is-rtl "
  data-garden-id="radios.radio_view"
  data-garden-version="version"
/>
`;

exports[`RadioView renders default styling 1`] = `
<div
  className="c-chk "
  data-garden-id="radios.radio_view"
  data-garden-version="version"
/>
`;
